sentry-openfeign/src/main/java/io/sentry/openfeign/SentryFeignClient.java

```diff
@@ -60,9 +60,16 @@ public Response execute(final @NotNull Request request, final @NotNull Request.O
         throw e;
       } finally {
         if (beforeSpan != null) {
-          span = beforeSpan.execute(span, request, response);
-        }
-        if (span != null) {
+          final ISpan result = beforeSpan.execute(span, request, response);
+
+          if (result == null) {
+            // span is dropped
+            span.getSpanContext().setSampled(false);
+          } else {
+            span = result;
+            span.finish();
+          }
+        } else {
```

Comment:
the logic here is a bit different than the one in the OkHttp, which one is correct?

Reply:
if I got it right, we dont finish the span in case its dropped

Reply:
Thanks, good catch.